components/movementsensor/movementsensor.go

```diff
@@ -22,44 +23,30 @@ func init() {
 		RPCServiceDesc:              &pb.MovementSensorService_ServiceDesc,
 		RPCClient:                   NewClientFromConn,
 	})
-
-	registerCollector("Position", func(ctx context.Context, ms MovementSensor, extra map[string]interface{}) (interface{}, error) {
-		type Position struct {
-			Lat float64
-			Lng float64
-		}
-		p, _, err := ms.Position(ctx, extra)
-		if err != nil {
-			return nil, err
-		}
-		return Position{Lat: p.Lat(), Lng: p.Lng()}, nil
-	})
-	registerCollector("LinearVelocity", func(ctx context.Context, ms MovementSensor, extra map[string]interface{}) (interface{}, error) {
-		v, err := ms.LinearVelocity(ctx, extra)
-		return v, err
-	})
-	registerCollector("AngularVelocity", func(ctx context.Context, ms MovementSensor, extra map[string]interface{}) (interface{}, error) {
-		v, err := ms.AngularVelocity(ctx, extra)
-		return v, err
-	})
-	registerCollector("CompassHeading", func(ctx context.Context, ms MovementSensor, extra map[string]interface{}) (interface{}, error) {
-		type Heading struct {
-			Heading float64
-		}
-		h, err := ms.CompassHeading(ctx, extra)
-		if err != nil {
-			return nil, err
-		}
-		return Heading{Heading: h}, nil
-	})
-	registerCollector("LinearAcceleration", func(ctx context.Context, ms MovementSensor, extra map[string]interface{}) (interface{}, error) {
-		v, err := ms.LinearAcceleration(ctx, extra)
-		return v, err
-	})
-	registerCollector("Orientation", func(ctx context.Context, ms MovementSensor, extra map[string]interface{}) (interface{}, error) {
-		v, err := ms.Orientation(ctx, extra)
-		return v, err
-	})
+	data.RegisterCollector(data.MethodMetadata{
+		API:        API,
+		MethodName: position.String(),
```

Comment:
Oh I see this is where we use the String function - same question about the API names applies. I notice that for other components the names match the Go method names and not the API method names so there is a precedent set but I'm wondering why. Especially if this affects how users register collectors in their configs (not actually sure that's true, I'd need to dig into the rdk code some more!)